Занятие 1:

Detrimon opened the pull request:
Задача 1: Вычисление факториала

Detrimon committed:
Занятие 1:
Задача 1: Вычисление факториала

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Вычислить факториал
+ факториал 3 равен 6 
+ факториал 1 равен 1
+ факториал 0 равен 1
+ факториал 5 равен 120 
```

Добавляю преподавателя (@stas-dolgachov) для код-ревью, посмотрим что он скажет :)

Detrimon committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20200727_detrimon

Detrimon committed:
Занятие 1:
Задача 2: Проверка пользователя на валидность его УЗ

jsru commented:
Пожалуйста, не меняйте «системные» файлы.
          Задача не может считаться решенной, если были изменены файлы тестов или другие «системные» файлы.
          Следующие файлы были изменены:
```diff
- 1-module/2-task/task.test.js — изменен
```

Detrimon committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20200727_detrimon

dolgachio commented on 1-module/2-task/task.test.js:
```diff
@@ -7,7 +7,15 @@ describe('1-module-2-task', () => {
   });
 
-  it('если меньше 4 символов то false', () => {
+  it('если меньше 4 символов, то false', () => {
```
Изменение тестов, к сожалению, приводит к тому, что задача не может быть принята :) 
Желательно вернуть все как было, я могу с этим помочь если с этим будут трудности

Detrimon replied:
Вроде бы вернул в начальное состояние, запятую тоже убрал :)

Detrimon committed:
Возвращение 2-task/task.test.js в исходное состояние

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

```diff
Вычислить факториал
+ факториал 5 равен 120 
+ факториал 0 равен 1
+ факториал 1 равен 1
+ факториал 3 равен 6 
```
```diff
Сделать первый символ заглавным
- должна делать первый символ заглавным
- должна работать для строки из одной буквы
- должна работать для пустой строки
```
```diff
Допишите функцию валидации имени пользователя
+ если имя корректное, то true
+ если содержит пробелы, то false
+ если меньше 4 символов то false
```

Detrimon committed:
Занятие 1:
Задание 3: Сделать первую букву заглавной

jsru commented:
@stas-dolgachov ping :)

dolgachio commented on 1-module/2-task/index.js:
```diff
@@ -11,6 +11,13 @@ function print(text) {
  */
 function isValid(name) {
   // ваш код...
+  return (name && name.length >= 4 && name.indexOf(' ') === -1) ? true : false;
```
Возможно здесь есть смысл не использовать тернарный оператор, а просто вернуть из функции выражение `name && name.length >= 4 && name.indexOf(' ') === -1`. При этом можно перевести `name` в логический тип, чтобы оно идеально правильно работало :)

Что думаете по этому поводу?

Detrimon replied:
Так?
Получилось так. Но если так, то работает, но мне кажется, что читать тяжело:
return !!name && !!(name.length >= 4) && !!(name.indexOf(' ') === -1);

dolgachio replied:
@Detrimon ну чтобы улучшить чтение, можно повыносить отдельные проверки в переменные. Я не против, если вы оставите так, как есть сейчас :)

Detrimon committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20200727_detrimon

Detrimon committed:
Занятие 1:
Задача 4: Спам-фильтр
Решение двумя способами: без RegExp и с RegExp.

Detrimon committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20200727_detrimon

jsru commented:
@stas-dolgachov ping :)

Detrimon committed:
Занятие 1:
Задача 5: Сокращение строки до N символов и добавление троеточия в конец при необходимости.

jsru commented:
@stas-dolgachov ping :)
